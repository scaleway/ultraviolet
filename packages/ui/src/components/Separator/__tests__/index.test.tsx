import { shouldMatchEmotionSnapshot } from '@utils/test'
import { describe, test } from 'vitest'
import { Separator } from '..'

describe('Separator', () => {
  test(`renders correctly with default props`, () =>
    shouldMatchEmotionSnapshot(<Separator />))
  test(`renders correctly with custom thickness`, () =>
    shouldMatchEmotionSnapshot(<Separator thickness={3} />))
  test(`renders correctly vertically`, () =>
    shouldMatchEmotionSnapshot(<Separator direction="vertical" />))
  test(`renders correctly horizontally`, () =>
    shouldMatchEmotionSnapshot(<Separator direction="horizontal" />))

  test(`renders correctly with custom color`, () =>
    shouldMatchEmotionSnapshot(<Separator color="primary" />))

  test(`renders correctly with custom color`, () =>
    shouldMatchEmotionSnapshot(<Separator sentiment="primary" />))

  test(`renders correctly with custom icon`, () =>
    shouldMatchEmotionSnapshot(<Separator icon="ray-top-arrow" />))

  test(`renders correctly with custom sentiment and icon`, () =>
    shouldMatchEmotionSnapshot(
      <Separator sentiment="primary" icon="ray-top-arrow" />,
    ))
  test(`renders correctly with custom icon vertically`, () =>
    shouldMatchEmotionSnapshot(
      <Separator direction="vertical" icon="ray-top-arrow" />,
    ))
  test(`renders correctly with custom icon horizontally`, () =>
    shouldMatchEmotionSnapshot(
      <Separator direction="horizontal" icon="ray-top-arrow" />,
    ))
})
