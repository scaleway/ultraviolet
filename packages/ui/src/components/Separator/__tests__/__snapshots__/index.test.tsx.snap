// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Separator > renders correctly horizontally 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  border: 0;
  width: auto;
  height: 1px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background-color: #e9eaeb;
}

<div
    data-testid="testing"
  >
    <hr
      aria-orientation="horizontal"
      class="emotion-0 emotion-1"
      role="separator"
    />
  </div>
</DocumentFragment>
`;

exports[`Separator > renders correctly vertically 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  border: 0;
  width: 1px;
  height: auto;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background-color: #e9eaeb;
}

<div
    data-testid="testing"
  >
    <hr
      aria-orientation="vertical"
      class="emotion-0 emotion-1"
      role="separator"
    />
  </div>
</DocumentFragment>
`;

exports[`Separator > renders correctly with custom color 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  border: 0;
  width: auto;
  height: 1px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background-color: #521094;
}

<div
    data-testid="testing"
  >
    <hr
      aria-orientation="horizontal"
      class="emotion-0 emotion-1"
      role="separator"
    />
  </div>
</DocumentFragment>
`;

exports[`Separator > renders correctly with custom color 2`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  border: 0;
  width: auto;
  height: 1px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background-color: #e9eaeb;
}

<div
    data-testid="testing"
  >
    <hr
      aria-orientation="horizontal"
      class="emotion-0 emotion-1"
      role="separator"
    />
  </div>
</DocumentFragment>
`;

exports[`Separator > renders correctly with custom icon 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 svg {
  fill: #e9eaeb;
}

.emotion-2 {
  margin: 0;
  border: 0;
  width: auto;
  height: 1px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background-color: #e9eaeb;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-4 {
  vertical-align: middle;
  fill: #3f4250;
  height: 24px;
  width: 24px;
  min-width: 24px;
  min-height: 24px;
}

.emotion-4 .fillStroke {
  stroke: #3f4250;
  fill: none;
}

<div
    data-testid="testing"
  >
    <div
      aria-orientation="horizontal"
      class="emotion-0 emotion-1"
      role="separator"
    >
      <hr
        class="emotion-2 emotion-3"
      />
      <svg
        class="emotion-4 emotion-5"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M8.25 2a.75.75 0 0 1 .75.75v8.69l1.22-1.22a.75.75 0 1 1 1.06 1.06l-2.5 2.5a.75.75 0 0 1-1.06 0l-2.5-2.5a.75.75 0 1 1 1.06-1.06l1.22 1.22V2.75A.75.75 0 0 1 8.25 2"
          fill-rule="evenodd"
        />
        <path
          d="M10.25 4a2 2 0 1 1-4 0 2 2 0 0 1 4 0"
        />
      </svg>
      <hr
        class="emotion-2 emotion-3"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Separator > renders correctly with custom icon horizontally 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 svg {
  fill: #e9eaeb;
}

.emotion-2 {
  margin: 0;
  border: 0;
  width: auto;
  height: 1px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background-color: #e9eaeb;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-4 {
  vertical-align: middle;
  fill: #3f4250;
  height: 24px;
  width: 24px;
  min-width: 24px;
  min-height: 24px;
}

.emotion-4 .fillStroke {
  stroke: #3f4250;
  fill: none;
}

<div
    data-testid="testing"
  >
    <div
      aria-orientation="horizontal"
      class="emotion-0 emotion-1"
      role="separator"
    >
      <hr
        class="emotion-2 emotion-3"
      />
      <svg
        class="emotion-4 emotion-5"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M8.25 2a.75.75 0 0 1 .75.75v8.69l1.22-1.22a.75.75 0 1 1 1.06 1.06l-2.5 2.5a.75.75 0 0 1-1.06 0l-2.5-2.5a.75.75 0 1 1 1.06-1.06l1.22 1.22V2.75A.75.75 0 0 1 8.25 2"
          fill-rule="evenodd"
        />
        <path
          d="M10.25 4a2 2 0 1 1-4 0 2 2 0 0 1 4 0"
        />
      </svg>
      <hr
        class="emotion-2 emotion-3"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Separator > renders correctly with custom icon vertically 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 svg {
  fill: #e9eaeb;
}

.emotion-2 {
  margin: 0;
  border: 0;
  width: 1px;
  height: auto;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background-color: #e9eaeb;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-4 {
  vertical-align: middle;
  fill: #3f4250;
  height: 24px;
  width: 24px;
  min-width: 24px;
  min-height: 24px;
}

.emotion-4 .fillStroke {
  stroke: #3f4250;
  fill: none;
}

<div
    data-testid="testing"
  >
    <div
      aria-orientation="vertical"
      class="emotion-0 emotion-1"
      role="separator"
    >
      <hr
        class="emotion-2 emotion-3"
      />
      <svg
        class="emotion-4 emotion-5"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M8.25 2a.75.75 0 0 1 .75.75v8.69l1.22-1.22a.75.75 0 1 1 1.06 1.06l-2.5 2.5a.75.75 0 0 1-1.06 0l-2.5-2.5a.75.75 0 1 1 1.06-1.06l1.22 1.22V2.75A.75.75 0 0 1 8.25 2"
          fill-rule="evenodd"
        />
        <path
          d="M10.25 4a2 2 0 1 1-4 0 2 2 0 0 1 4 0"
        />
      </svg>
      <hr
        class="emotion-2 emotion-3"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Separator > renders correctly with custom sentiment and icon 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 svg {
  fill: #521094;
}

.emotion-2 {
  margin: 0;
  border: 0;
  width: auto;
  height: 1px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background-color: #521094;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-4 {
  vertical-align: middle;
  fill: #3f4250;
  height: 24px;
  width: 24px;
  min-width: 24px;
  min-height: 24px;
}

.emotion-4 .fillStroke {
  stroke: #3f4250;
  fill: none;
}

<div
    data-testid="testing"
  >
    <div
      aria-orientation="horizontal"
      class="emotion-0 emotion-1"
      role="separator"
    >
      <hr
        class="emotion-2 emotion-3"
      />
      <svg
        class="emotion-4 emotion-5"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M8.25 2a.75.75 0 0 1 .75.75v8.69l1.22-1.22a.75.75 0 1 1 1.06 1.06l-2.5 2.5a.75.75 0 0 1-1.06 0l-2.5-2.5a.75.75 0 1 1 1.06-1.06l1.22 1.22V2.75A.75.75 0 0 1 8.25 2"
          fill-rule="evenodd"
        />
        <path
          d="M10.25 4a2 2 0 1 1-4 0 2 2 0 0 1 4 0"
        />
      </svg>
      <hr
        class="emotion-2 emotion-3"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Separator > renders correctly with custom thickness 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  border: 0;
  width: auto;
  height: 3px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background-color: #e9eaeb;
}

<div
    data-testid="testing"
  >
    <hr
      aria-orientation="horizontal"
      class="emotion-0 emotion-1"
      role="separator"
    />
  </div>
</DocumentFragment>
`;

exports[`Separator > renders correctly with default props 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  border: 0;
  width: auto;
  height: 1px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background-color: #e9eaeb;
}

<div
    data-testid="testing"
  >
    <hr
      aria-orientation="horizontal"
      class="emotion-0 emotion-1"
      role="separator"
    />
  </div>
</DocumentFragment>
`;
